nuget/helpers/lib/NuGetUpdater/DotNetPackageCorrelation/Model/PackageMapper.cs

```diff
@@ -0,0 +1,68 @@
+using Semver;
+
+namespace DotNetPackageCorrelation;
+
+public class PackageMapper
+{
+    public RuntimePackages RuntimePackages { get; }
+
+    private PackageMapper(RuntimePackages runtimePackages)
+    {
+        RuntimePackages = runtimePackages;
+    }
+
+    /// <summary>
+    /// Find the version of <paramref name="candidatePackageName"/> that shipped at the same time as
+    /// &quot;<paramref name="packageName"/>/<paramref name="packageVersion"/>&quot;.
+    /// </summary>
+    public SemVersion? GetPackageVersionThatShippedWithOtherPackage(string packageName, SemVersion packageVersion, string candidatePackageName)
+    {
+        var runtimeVersion = GetRuntimeVersionFromPackage(packageName, packageVersion);
+        if (runtimeVersion is null)
+        {
+            // no runtime found that contains the package
+            return null;
+        }
+
+        var candidateRuntimeVersions = RuntimePackages.Runtimes.Keys
+            .Where(v => v.Major == runtimeVersion.Major)
```

Comment:
Any reason not to combine these where clauses with an `&&`?

Reply:
Easier to debug by commenting out single lines.